src/lucky/memoizable.cr

```diff
@@ -25,15 +25,31 @@ module Lucky::Memoizable
       raise "All arguments must have an explicit type for memoized methods" if method_def.args.any? &.is_a?(Nop)
     %}
 
-    @__memoized_{{method_def.name}} : Tuple(
+    {%
+      is_predicate = false
+      is_bang = false
```

Comment:
If these are replaced with just a `special_ending = nil` variable and the two checks set it the way it's supposed to, you don't have to do any of the if/else's later on it'd just be

```crystal
def {{ safe_method_name }}__uncached{{ special_ending }}(
```